### PR TITLE
Adding non-neural baselines

### PR DESCRIPTION
**Pull request by neel-alex**
Currently: Naive Bayes and SVM, but adding Random Forest or AdaBoost would be trivial.

Does better than random in LOO testing, but not by much for SVM. Haven't tuned any hyperparameters.

### CODE REVIEW

**Comment by neel-alex** [comment]
Refactored so that the non-neural approaches can pass as the GPT-3 classifier. raft_predict now should smoothly accept both by only changing the configs (though they're growing a little complicated). 

Added AdaBoost. Takes substantially longer than NaiveBayes: 4 minutes to run on the entire test dataset vs 40 seconds.

File: src/raft_baselines/classifiers/random_classifier.py [hunk 1]
```diff
@@ -13,7 +13,13 @@ def __init__(
         super().__init__(training_data)
         random.seed(seed)
 
-    def classify(self, target: Mapping[str, str], random_seed: Optional[int] = None) -> Mapping[str, float]:
+    def train(self, **classifier_kwargs):
```
Comment (uvafan) [inline src/raft_baselines/classifiers/random_classifier.py]: we don't need this train method anymore right?

File: src/raft_baselines/scripts/raft_predict.py [hunk 2]
```diff
@@ -42,6 +42,23 @@ def base_config():
         # change to davinci to replicate results from the paper
         # "engine": "ada",
     }
+    if classifier_name in ('NaiveBayesClassifier', 'SVMClassifier', 'AdaBoostClassifier'):
```
Comment (uvafan) [inline src/raft_baselines/scripts/raft_predict.py:45]: test instead if it's an `instanceof` `NonNeuralClassifier`?

File: src/raft_baselines/classifiers/non_neural_classifier.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+from typing import Mapping, Optional, Dict
+from abc import abstractmethod
+
+import datasets
+from sklearn.feature_extraction.text import CountVectorizer
+
+from raft_baselines.classifiers.classifier import Classifier
+
+
+class NonNeuralClassifier(Classifier):
```
Comment (uvafan) [inline src/raft_baselines/classifiers/non_neural_classifier.py]: Maybe a more precise name would be `NGramsClassifier`?

File: src/raft_baselines/scripts/test_naive_bayes.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+from raft_baselines.classifiers import NaiveBayesClassifier
+from raft_baselines.classifiers import RandomClassifier
+
+train = datasets.load_dataset(
+    "ought/raft", "banking_77", split="train"
+)
+
+classifier = NaiveBayesClassifier(train)
+
+print(classifier.classify({"Paper title": "CNN research", "Impact statement": "test2"}))
```
Comment (uvafan) [inline src/raft_baselines/scripts/test_naive_bayes.py:11]: nit: mismatch between examples and dataset loaded above

**Review by uvafan — APPROVED** [review]
Thanks for implementing this, left a few minor comments to address before merging

**Pull request merged**